projects/backend/controllers/render.ts

```diff
@@ -16,24 +21,75 @@ const fetchRenderedArticle = async (url: string) => {
     }
 }
 
-// TODO: this needs a test once we're happy with the correct format for the paths
-const replaceImageUrls = (html: string): string => {
-    return html.replace(/https:\/\/i.guim.co.uk\/img\//g, `../media/`)
+const fetchCapiArticle = async (
+    internalPageCode: number,
+    capi: CAPIEndpoint,
+): Promise<ArrayBuffer | undefined> => {
+    try {
+        const endpoint = generateCapiEndpoint([internalPageCode], capi)
+        const headers =
+            capi === 'preview' ? await getPreviewHeaders(endpoint) : {}
+        console.log('Making CAPI query', endpoint)
+        console.log('Debug link:', endpoint.replace(/thrift/g, 'json'))
+        const resp = await attempt(fetch(endpoint, { headers }))
+
+        if (hasFailed(resp)) throw new Error('Could not connect to CAPI.')
+
+        if (resp.status != 200) {
+            console.warn(
+                `Non 200 status code: ${resp.status} ${resp.statusText}`,
+            )
+        }
+
+        return await resp.arrayBuffer()
+    } catch (error) {
+        console.warn(
+            `Failed to fetch article from capi (${capi}) for internalPageCode=${internalPageCode}`,
+        )
+        return undefined
+    }
 }
 
 export const renderController = async (req: Request, res: Response) => {
-    const path = req.params.path
-    const renderingUrl = `${process.env.APPS_RENDERING_URL}${path}?editions`
-    console.log(`Fetching ${renderingUrl} from apps rendering`)
-    const renderResponse = await fetchRenderedArticle(renderingUrl)
+    const pageCode = req.params.internalPageCode
+    //The article we are looking for could be in any environment, need to
+    //make an attempt to fetch from all three environment
+    console.log(`Trying to fetch article from capi (live)`)
+    let capiResponse = await fetchCapiArticle(pageCode, 'live')
+    if (!capiResponse) {
+        console.log(
+            `Article was not found in capi (live), trying to fetch from printsent`,
+        )
+        capiResponse = await fetchCapiArticle(pageCode, 'printsent')
+    }
+    if (!capiResponse) {
+        console.log(
+            `Article was not found in capi (printsent), trying to fetch from preview`,
+        )
+        capiResponse = await fetchCapiArticle(pageCode, 'preview')
+    }
+
+    if (!capiResponse) {
+        const message = `Failed to fetch article from capi (print, live & preview) for ${pageCode}`
```

Comment:
Maintain the order of the API attempts in the log statement

`Failed to fetch article from capi (live, printsent and preview) for ${pageCode}`